File: data/gpt-4o/readable_format_HPEC/cpp/file292.cpp
void kernel_trmm(int ni, double alpha, double a[100][100], double b[100][100])
{
    int i, j, k;
#pragma scop
 for (i = 1; i < ni; i++)
 for (j = 0; j < ni; j++)
 for (k = 0; k < i; k++)
 b[j][i] = b[j][i] + (alpha * a[k][i] * b[k][j]);
#pragma endscop
}

File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file292.f90
subroutine kernel_trmm(ni, alpha, a, b)
implicit none

DATA_TYPE, dimension(ni, ni) :: a
DATA_TYPE, dimension(ni, ni) :: b
DATA_TYPE :: alpha
integer :: ni
integer :: i, j, k

!$pragma scop
do i = 2, _PB_NI
do j = 1, _PB_NI
do k = 1, i - 1
b(j, i) = b(j, i) + (alpha * a(k, i) * b(k, j))
end do
end do
end do
!$pragma endscop
end subroutine
